### PR TITLE
Javascript javascript2 week2

### PR DESCRIPTION
**Pull request by shravanipagilla**

### CODE REVIEW

File: javascript/javascript2/week2/working-with-movies.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...rand bleu","year": 1988,"rating": 7.6,"votes": 41252,"running_times": 10080},{"title": "Le locataire","year": 1976,"rating": 7.8,"votes": 30919,"running_times": 7560},{"title": "Le mépris","year": 1963,"rating": 7.7,"votes": 22380,"running_times": 6180},{"title": "Le procès","year": 1962,"rating": 7.8,"votes": 15401,"running_times": 7140},{"title": "Le salaire de la peur","year": 1953,"rating": 8.2,"votes": 40345,"running_times": 9360},{"title": "Le violon rouge","year": 1998,"rating": 7.7,"votes": 28394,"running_times": 8400},{"title": "Lean on Me","year": 1989,"rating": 7.4,"votes": 15573,"running_times": 6480},{"title": "Leap Year","year": 2010,"rating": 6.5,"votes": 78668,"running_times": 6000},{"title": "Leap of Faith","year": 1992,"rating": 6.1,"votes": 10526,"running_times": 6480},{"title": "Leatherface: Texas Chainsaw Massacre III","year": 1990,"rating": 5.2,"votes": 11025,"running_times": 5100},{"title": "Leatherheads","year": 2008,"rating": 6,"votes": 28907,"running_times": 6840},{"title": "Leaves of Grass","year": 2009,"rating": 6.5,"votes": 24966,"running_times": 6300},{"title": "Leaving Las Vegas","year": 1995,"rating": 7.6,"votes": 100091,"running_times": 6660},{"titl
```
Comment (dsamourkasidis) [inline javascript/javascript2/week2/working-with-movies.js]: You could also do directly:
```
const shortTitleMovies = movies.filter((movie) => movie.title < 5);
```

Comment (dsamourkasidis) [inline javascript/javascript2/week2/working-with-movies.js]: Try to use filter here as well similar as above!

Comment (dsamourkasidis) [inline javascript/javascript2/week2/working-with-movies.js]: Filter is used for filtering data, so .map is a better fit to add a new key here:
```
extraKey = movies.map((movie) => {
  if (movie.rating <= 4) {
    movie.tag = "Bad";
  }
  if (movie.rating > 4 && movie.rating < 7) {
    movie.tag = "Average";
  }
  if (movie.rating >= 7) {
    movie.tag = "Good";
  }
```

Comment (dsamourkasidis) [inline javascript/javascript2/week2/working-with-movies.js]: Good job using .toLowerCase(). You could also avoid the if/else if statement using the OR(||) operator:
```
movies.filter(m => m.title.toLowerCase().includes("surfer") || m.title.toLowerCase().includes("alien") || m.title.toLowerCase().includes("benjamin")).length;
```

Comment (dsamourkasidis) [inline javascript/javascript2/week2/working-with-movies.js]: Here's a solution using .reduce:
```
const countMovies = movies.reduce(
  function (accumulator, currentItem) {
    if (currentItem.rating >= 7) accumulator.goodMovies += 1;
    if (currentItem.rating >= 4 && currentItem.rating < 7) accumulator.averageMovies += 1;
    if (currentItem.rating < 4) accumulator.badMovies += 1;
    return accumulator;
  },
  {
    goodMovies: 0,
    averageMovies: 0,
    badMovies: 0,
  },
);
```
The hint here is that .[reduce ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)accepts an initialValue argument

**Review by dsamourkasidis — COMMENTED** [review]
Very good job 👍 
I left some small comments you can go through and fix/improve

File: javascript/javascript2/week2/working-with-movies.js (hunk 1 above)
Comment (shravanipagilla) [reply javascript/javascript2/week2/working-with-movies.js:6]: Please look into line 6

Comment (shravanipagilla) [reply javascript/javascript2/week2/working-with-movies.js:22]: PLEASE REFER LINE NUMBER..
first line is for the data.

**Pull request merged**